app/controllers/books_controller.rb

```diff
@@ -0,0 +1,55 @@
+class BooksController < ApplicationController
+  def index
+    @books = Book.all
+  end
+
+  def show
+    @book = Book.find(params[:id])
+  end
+
+  def edit
+    @book = Book.find(params[:id])
+  end
+
+  def new
+    @book = Book.new
+  end
+
+  def create
+    @book = Book.new(book_params)
+    if @book.save
+      redirect_to books_path
+    else
+      render :new
+    end
+  end
+
+  def update
+    id = params[:id]
+    book = Book.find(id)
+    book.update_attributes(book_params)
+    redirect_to book_path(params[:id])
+  end
+
+  def destroy
+    Book.destroy(params[:id])
+    redirect_to books_path
+  end
+
+  def upvote
```

Comment:
Since the logic in this method is affecting the object, it would be more appropriate as a model method which you then call from the controller.